### ISSUE TITLE
ListView.Items.RemoveAt causes NullReferenceException

### ISSUE THREAD

**Issue by 2mik** [dotnet/winforms#6230]
* .NET Core Version: 
6.0.0

* Have you experienced this same bug with .NET Framework?: 
Classic .NET - No
.NET 5 - No
.NET 6 - Yes

**Problem description:**
Deleting a selected item from ListView causes NullReferenceException on calling ListView.Items.RemoveAt(index).
The bug reproduces stable in my app in the following conditions:
1. The item being deleted must be selected. If it's not selected, no error.
2. To workaround the error, we can call ListView.Focus() before delete.
3. ListView is a Details view.

The code that fails:
`

    // listView1.Focus(); // helps to avoid error

    listView1.Items.RemoveAt(0); // error is here if item 0 is selected

    MessageBox.Show("a"); // shows after error
`

Despite the error, the item is deleted.

Error stack:
System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Windows.Forms.ListViewItem.get_AccessibilityObject()
   at System.Windows.Forms.ListView.ListViewAccessibleObject.GetFocus()
   at System.Windows.Forms.ListView.OnGotFocus(EventArgs e)
   at System.Windows.Forms.Control.WmSetFocus(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ListView.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WM msg, IntPtr wparam, IntPtr lparam)

**Expected behavior:**
No error in the above case. Because it worked well before .NET 6.

**Minimal repro:**
I couldn't create a small app to reproduce the error, because there is no error in a simple app. However, I hope that a developer who works daily on WinForms, can find the possible mistake in ListViewItem.get_AccessibilityObject()
Edit: the test app is attached below.

**Comment by 2mik** [comment]
May be add ? after Group in this line
ListView owningListView = listView ?? Group**?**.ListView ?? throw new InvalidOperationException(nameof(listView));
in [this line](https://github.com/dotnet/winforms/blob/4c1f44c52a62bf4dafc178fc4a2f900c7c3fb9fa/src/System.Windows.Forms/src/System/Windows/Forms/ListViewItem.cs#L238)

**Comment by 2mik** [comment]
I've managed to create an app to reproduce the error:
[WinFormsApp_VS2022.zip](https://github.com/dotnet/winforms/files/7602653/WinFormsApp_VS2022.zip)

1. Run the app.
2. Press the "button1"
3. Press the Add button.
4. Click the white space of ListView to deselect item.
5. Click the 2nd item.
6. Click the Delete button.
7. You should get a described exception.

**Comment by RussKie** [comment]
@vladimir-krestov @SergeySmirnov-Akvelon please investigate

**Comment by RussKie** [comment]
@SergeySmirnov-Akvelon it would be great to finalise the fixes ASAP, 6.0.3 branches are open now.

**Comment by MelonWang1** [comment]
Verified this issue with .NET SDK 6.0.201, this issue is fixed.
![pull65](https://user-images.githubusercontent.com/94418985/155291265-773758c2-5f64-492b-af7b-3ab97a76fa01.gif)

**Comment by MelonWang1** [comment]
Verified with .NET SDK 7.0.100-preview.2.22152.5 build, this issue is fixed.
